src/Sylius/Bundle/ShopBundle/gulpfile.babel.js

```diff
@@ -0,0 +1,176 @@
+      required: false,
+    },
+    nodeModulesPath: {
+      description: '<path> path to node_modules directory',
+      type: 'string',
+      requiresArg: true,
+      required: true,
+    },
+  });
+
+const env = process.env.GULP_ENV;
+const rootPath = upath.normalizeSafe(argv.rootPath);
+const shopRootPath = upath.joinSafe(rootPath, 'shop');
+const vendorPath = upath.normalizeSafe(argv.vendorPath || '.');
+const vendorShopPath = vendorPath === '.' ? '.' : upath.joinSafe(vendorPath, 'ShopBundle');
+const vendorUiPath = vendorPath === '.' ? '../UiBundle/' : upath.joinSafe(vendorPath, 'UiBundle');
+const nodeModulesPath = upath.normalizeSafe(argv.nodeModulesPath);
+
+const paths = {
+  shop: {
+    js: [
+      upath.joinSafe(nodeModulesPath, 'jquery/dist/jquery.min.js'),
+      upath.joinSafe(nodeModulesPath, 'semantic-ui-css/semantic.min.js'),
+      upath.joinSafe(nodeModulesPath, 'lightbox2/dist/js/lightbox.js'),
+      upath.joinSafe(vendorUiPath, 'Resources/private/js/**'),
+      upath.joinSafe(vendorShopPath, 'Resources/private/js/**'),
+    ],
+    sass: [
+      upath.joinSafe(vendorUiPath, 'Resources/private/sass/**'),
+      upath.joinSafe(vendorShopPath, 'Resources/private/sass/**'),
+    ],
+    css: [
+      upath.joinSafe(nodeModulesPath, 'semantic-ui-css/semantic.min.css'),
+      upath.joinSafe(nodeModulesPath, 'lightbox2/dist/css/lightbox.css'),
+      upath.joinSafe(vendorUiPath, 'Resources/private/css/**'),
+      upath.joinSafe(vendorShopPath, 'Resources/private/css/**'),
+      upath.joinSafe(vendorShopPath, 'Resources/private/scss/**'),
+    ],
+    img: [
+      upath.joinSafe(vendorUiPath, 'Resources/private/img/**'),
+      upath.joinSafe(vendorShopPath, 'Resources/private/img/**'),
+    ],
+  },
+};
+
+const sourcePathMap = [
+  {
+    sourceDir: upath.relative('', upath.joinSafe(vendorShopPath, 'Resources/private')),
+    destPath: '/SyliusShopBundle/',
+  },
+  {
+    sourceDir: upath.relative('', upath.joinSafe(vendorUiPath, 'Resources/private')),
+    destPath: '/SyliusUiBundle/',
+  },
+  {
+    sourceDir: upath.relative('', nodeModulesPath),
+    destPath: '/node_modules/',
+  },
+];
+
+const mapSourcePath = function mapSourcePath(sourcePath /* , file */) {
+  const match = sourcePathMap.find(({ sourceDir }) => (
+    sourcePath.substring(0, sourceDir.length) === sourceDir
+  ));
+
+  if (!match) {
+    return sourcePath;
+  }
+
+  const { sourceDir, destPath } = match;
+
+  return upath.joinSafe(destPath, sourcePath.substring(sourceDir.length));
+};
+
+export const buildShopJs = function buildShopJs() {
+  return gulp.src(paths.shop.js, { base: './' })
+    .pipe(gulpif(env !== 'prod', sourcemaps.init()))
+    .pipe(concat('app.js'))
+    .pipe(gulpif(env === 'prod', uglify()))
+    .pipe(gulpif(env !== 'prod', sourcemaps.mapSources(mapSourcePath)))
+    .pipe(gulpif(env !== 'prod', sourcemaps.write('./')))
+    .pipe(gulp.dest(upath.joinSafe(shopRootPath, 'js')))
+    .pipe(livereload());
```

Comment:
I've added the `livereload()` call here because it seemed to be missing. Let me know if it was intentionally omitted.